project/src/main/java/com/google/autograder/data/Database.java

```diff
@@ -1,151 +1,337 @@
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.datastore.Query.Filter;
 import com.google.appengine.api.datastore.PreparedQuery;
-import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.Query.SortDirection;
-import com.google.appengine.api.datastore.Key;
-import com.google.appengine.api.datastore.Query.CompositeFilter;
-import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
-import com.google.appengine.api.datastore.Query.Filter;
 import com.google.appengine.api.datastore.Query.FilterOperator;
 import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
 
 
+import java.util.Iterator;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+// import java.net.URL;
+// import java.io.IOException;
+// import java.io.BufferedReader;
+// import java.io.InputStreamReader;
+// import java.net.HttpURLConnection;
+// import javax.servlet.http.HttpServlet;
+// import java.nio.charset.StandardCharsets;
+// import javax.servlet.annotation.WebServlet;
+// import java.io.UnsupportedEncodingException;
+// import org.json.simple.parser.ParseException;
+// import javax.servlet.http.HttpServletRequest;
+// import javax.servlet.http.HttpServletResponse;
+// import com.google.appengine.api.datastore.Query;
+// import com.google.appengine.api.datastore.Entity;
+// import com.google.autograder.servlets.helpers.API;
+// import com.google.autograder.servlets.helpers.Services;
+// import com.google.appengine.api.datastore.Query.Filter;
+// import com.google.appengine.api.datastore.PreparedQuery;
+// import com.google.appengine.api.datastore.Query.SortDirection;
+// import com.google.appengine.api.datastore.Query.FilterOperator;
+// import com.google.autograder.servlets.auth.AccessTokenResponse;
+// import com.google.appengine.api.datastore.Query.FilterPredicate;
+
 /** Class containing basic database functionalities. */
-public class Database {
+public final class Database {
+
+    private static final UserService USER_SERVICE = UserServiceFactory.getUserService();
+    private static final DatastoreService DATA_STORE = DatastoreServiceFactory.getDatastoreService();
+    private static final Gson GSON = new Gson();
+
+    public static final List<String> ANSWER_GRADING_STATUSES = Arrays.asList("NOT_GRADED", "GRADED");
+
+    public static UserService getUserService() {
```

Comment:
Typically exposing the underlying variables means there's a flaw with the design of the class. This class should be hiding the fact that it needs to use UserService or DatastoreService at all, and callers shouldn't need to access either one. An example of good methods here are getCurrentUserEmail and getCurrentUserID, which returns exactly what the caller requires and does not expose the fact that it needs to call UserService.

An example of a method that does not abstract well is query, since it returns a PreparedQuery, which presumably needs to be passed to the DatastoreService. A better method signature would be:
```Iterable<Entity> query(Query query) {}```
so that callers won't have to access the underlying DatastoreService